### ISSUE TITLE
Saving an experiment doesn't copy the Analog Output Function

### ISSUE THREAD

**Issue by floesche** [JaneliaSciComp/G4_Display_Tools#77]
How to replicate:

- Start Experiment Designer
- Create a simple protocol with at least one Analog Output Function
- File -> Save As
  - choose new directory (in my case "c:\matlabroot\MyExperiment01"
  - Choose "MyExperiment01.g4p" as a file name
  - Click "Save"

![image](https://github.com/JaneliaSciComp/G4_Display_Tools/assets/1568893/35fc8711-9371-4759-9a28-9ecf211700cc)

The resulting folder `06-22-23_13-16-15` contains the subfolders `Analog Output Functions`, `Functions`, and `Patterns` along with `MyExperiment01.g4p` and `currentExp.mat`. `Functions` and `Patterns` have the `*.mat` and `*.pfn` / `*.pat` files, but `Analog Output Functions` is empty.

**Comment by floesche** [comment]
In a second test with just one trial and 2 differently named AO functions, the export worked.

So maybe the problem seen above is related to either the frame index or using the same name for the Position and the Analog function? 

Needs more investigation.

**Comment by taylorlm88** [comment]
In the first attempt when you had trouble saving, the AO function name is the same as the position function name - but is it actually the same file? Is the binary that goes with the function a .pfn or .afn file? Does it have the prefix fun or ao?

**Comment by floesche** [comment]
No, it was not the same file. I created a Position function with that name first, which created a `*pfn` file in the `Functions` folder. Then switched over to AO functions and created an AO function with the same name, which created a `*afn` file in the `Analog Output Functions` folder.

In the designer I loaded the whole folder, which gave me different functions (and renderings) for functions/AO functions.

**Comment by taylorlm88** [comment]
This happened to me yesterday when testing out the AO index display in the Conductor. If I imported the functions and created a new experiment with them and saved, it worked fine. But if I opened an existing protocol with AO functions, changed something, then saved it, the AO functions did not export. Will look into it this week.

**Comment by taylorlm88** [comment]
Updating to add that I tried this on my computer (when it happened to me last week I was on Eyal's old computer) and I couldn't recreate the issue. No matter how I did it, the AO functions exported just fine. Will see if I can re-create on the other computer and if so, i'll investigate the difference.

**Comment by floesche** [comment]
Probably fixed
